### PR TITLE
You can now link fishing portal generators to other fishing spots.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
You can now interact with the fishing portal generator with a multitool to load it in the buffer and subsequently link fishing spots to it by tapping them with the same multitool. The maximum number of fishing spots that can be linked at once depends on the tier of the matter bins of the machinery. Normally, while you can link fishing spots from other z-levels, they can only be activated if they're on the same z-level as the machinery (or if both are on station for multi-z stations). This limitation can be bypassed by upgrading the machinery with a tier 3 or higher capacitator.

While it's possible, I'm not spriting new fishing portal overlays and icons for the radial menu for every fish source out there (yet). The code is enough work for now.

This also comes with a unit test, because there is no such thing as too many unit tests for fishing.

## Why It's Good For The Game
Fish portal generators are designed to let players fish a different bunch of things while being able to be moved wherever you like, unlike a lake or an ocean, with all the comfort of being able to able to catch fish from distant locations. Allowing players to link other fishing spots to it fits its design. It also means that you can go out and explore, find more fishing spots and then return to the station without having to detach yourself from the ongoing round for several more minutes.

## Changelog

:cl:
add: You can now link fishing portal generators to other fishing spots with a multitool. The number of fishing spots that can be linked at once and whether the link can be activated from different z levels depends on the tier of the stock parts it's built with.
/:cl:

### CODE REVIEW

**Comment by OuroborusReporter** [comment]
God bless, now the Moisture Trap -> Kitchen sludgefish taco supply chain is more efficient

**Comment by Ghommie** [comment]
It actually comes from toilets, but ok.

File: code/modules/fishing/fishing_portal_machine.dm [hunk 1]
```diff
@@ -11,6 +11,46 @@
...
+	return ..()
+
+///Higher tier parts let you link to more fishing spots at once and eventually let you connect through different zlevels.
+/obj/machinery/fishing_portal_generator/RefreshParts()
+	. = ..()
+	max_fishing_spots = 0
+	long_range_link = FALSE
+	for(var/datum/stock_part/matter_bin/matter_bin in component_parts)
+		max_fishing_spots += matter_bin.tier * 0.5
+	max_fishing_spots = round(max_fishing_spots)
```
Comment (ninjanomnom) [inline code/modules/fishing/fishing_portal_machine.dm:44]: Did you intend to make it so tier 1 has 0 fishing spots? `round(x)` is the same as `floor(x)`. You probably want the following here
```suggestion
	max_fishing_spots = ROUND_UP(max_fishing_spots)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Comment by ninjanomnom** [comment]
It looks like the CI failure here is the one solved after auto merge merged that failing CI previously so I'm going to just...

**Pull request merged**